ts/index.ts

```diff
@@ -6,18 +6,34 @@ import { StorageRegistry } from 'storex/ts'
 import * as backend from 'storex/ts/types/backend'
 import { augmentCreateObject } from 'storex/ts/backend/utils'
 import { getDexieHistory, getTermsIndex } from './schema'
-import { DexieMongoify } from './types'
+import { DexieMongoify, DexieSchema } from './types'
 import { IndexDefinition, CollectionField, CollectionDefinition } from 'storex/ts/types';
 import { StorageBackendFeatureSupport } from 'storex/ts/types/backend-features';
+import { UnimplementedError, InvalidOptionsError } from 'storex/ts/types/errors';
 
 export interface IndexedDbImplementation {
     factory : IDBFactory
     range : new () => IDBKeyRange
 }
 
 export type Stemmer = (text : string) => Set<string>
+export type SchemaPatcher = (schema: DexieSchema[]) => DexieSchema[]
+
+interface Props {
+    dbName : string
+    stemmer? : Stemmer
+    idbImplementation? : IndexedDbImplementation
+    /**
+     * An optional function to run the generated Dexie schemas through to
+     * afford changing them independently of the storex registry. Identity
+     * function by default.
+     **/
+    schemaPatcher? : SchemaPatcher
+}
 
 export class DexieStorageBackend extends backend.StorageBackend {
+    static DEF_SCHEMA_PATCHER: SchemaPatcher = f => f
```

Comment:
Just a matter of taste, but I kind of dislike static members for default arguments. Could you move this just below the SchemaPatcher type and name it NoopSchemaPatcher, which you can use as the default argument? Just reads a bit nicer, and you can see in a glance what the default behavior is.

Reply:
Have updated that. Thanks for letting me know; will keep in mind in the future. Although I ended up naming it `IdentitySchemaPatcher`, as I feel a noop function shouldn't return anything.

Reply:
True, I was also thinking about that name, couldn't really decide  :)